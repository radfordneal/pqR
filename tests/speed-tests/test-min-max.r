source("time.r")
 
test.name <- "min-max.real"

ar1 <- seq(0.001,1,length=1000)
ar2 <- rev(ar1)
ar3 <- ar1*ar2

f <- test.cmp (function () {for (i in 1:500000) b <- max(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:50000) b <- pmax(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])

f <- test.cmp (function () {for (i in 1:500000) b <- min(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:50000) b <- pmin(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])

ar1 <- as.integer(100*ar1)
ar2 <- as.integer(100*ar2)
ar3 <- as.integer(100*ar3)

test.name <- "min-max.int"

f <- test.cmp (function () {for (i in 1:500000) b <- max(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:50000) b <- pmax(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])

f <- test.cmp (function () {for (i in 1:500000) b <- min(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:50000) b <- pmin(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])

test.name <- "min-max.string"

ar1 <- as.character(100+ar1)
ar2 <- as.character(100+ar2)
ar3 <- as.character(100+ar3)

f <- test.cmp (function () {for (i in 1:50000) b <- max(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:20000) b <- pmax(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])

f <- test.cmp (function () {for (i in 1:50000) b <- min(ar3); b})
sys.time(b<-f()); print(b)

f <- test.cmp (function () {for (i in 1:20000) b <- pmin(ar1,ar2); b})
sys.time(b<-f()); print(b[c(1,2,999,1000)])
