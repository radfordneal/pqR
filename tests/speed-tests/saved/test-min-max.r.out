
---- min-max.real 

Time:
[1] 0.2505
Time:
[1] 1.000 0.999 0.999 1.000
Time:
[1] 0.001
Time:
[1] 0.001 0.002 0.002 0.001

---- min-max.int 

Time:
[1] 25
Time:
[1] 100  99  99 100
Time:
[1] 0
Time:
[1] 0 0 0 0

---- min-max.string 

Time:
[1] "125"
Time:
[1] "200" "199" "199" "200"
Time:
[1] "100"
Time:
[1] "100" "100" "100" "100"
